web_pwa_oca/static/src/js/worker/libs/class.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Improved John Resig's inheritance, based on:
+ *
+ * Simple JavaScript Inheritance
+ * By John Resig http://ejohn.org/
+ * MIT Licensed.
+ *
+ * Adds "include()"
+ *
+ * Defines The Class object. That object can be used to define and inherit classes using
+ * the extend() method.
+ *
+ * Example::
+ *
+ *     var Person = Class.extend({
+ *      init: function(isDancing){
+ *         this.dancing = isDancing;
+ *       },
+ *       dance: function(){
+ *         return this.dancing;
+ *       }
+ *     });
+ *
+ * The init() method act as a constructor. This class can be instanced this way::
+ *
+ *     var person = new Person(true);
+ *     person.dance();
+ *
+ *     The Person class can also be extended again:
+ *
+ *     var Ninja = Person.extend({
+ *       init: function(){
+ *         this._super( false );
+ *       },
+ *       dance: function(){
+ *         // Call the inherited version of dance()
+ *         return this._super();
+ *       },
+ *       swingSword: function(){
+ *         return true;
+ *       }
+ *     });
+ *
+ * When extending a class, each re-defined method can use this._super() to call the previous
+ * implementation of that method.
+ *
+ * @class Class
+ */
+function OdooClass(){}
```

Comment:
This is copied from upstream, right? Why is this needed? :thinking:

Reply:
Yes is just a copy but without "require" system. I can't use 'Odoo Bootstrap' here because depends on jQuery and jQuery uses DOM operations that are incompatible with ServiceWorkers.

ServiceWorkers are a piece separated from the page and doesn't have direct access to the DOM.

Reply:
Humm I see...

:warning: I know I might be saying something stupid!

You probably thought about this also, but I'm wondering... Could you mock the `require()` function instead? I'd say a simple implementation of that would be much less code and would give more reusability for other modules of Odoo.

In any case, some docs please explaining why all this is needed.